Move `OptionSet` to its own namespace

ahoppen opened the pull request:
The type was colliding with the `OptionSet` type in the Swift stdlib when using Swift from C++ (ie. when we were importing a the generated `<SwiftModule>-Swift.h` bridging header.

ahoppen commented:
@swift-ci Please smoke test

rintaro commented on include/swift/Basic/OptionSet.h:
```diff
@@ -151,7 +152,7 @@ class OptionSet {
                 "operator| should produce an OptionSet");
 };
-
+} // end namespace optionset
```
Does `using namespace` has any effect in Swift? If not, how about
```suggestion
} // end namespace optionset
using namespace optionset;
```
And that's it (with some doc-comment for this weird hack).

egorzhdan replied:
`using namespace` is ignored by Swift when it imports C++ headers.

ahoppen replied:
Oh, nice idea. I hadn’t thought of that. I tried `using OptionSet = optionset::OptionSet`, which didn’t work but this is much cleaner (and doesn’t require any downstream changes).

ahoppen committed:
Move `OptionSet` to its own namespace

The type was colliding with the `OptionSet` type in the Swift stdlib when using Swift from C++.

ahoppen commented:
@swift-ci Please smoke test
